dynawo/sources/Solvers/Common/DYNSolverCommon.h

```diff
@@ -45,6 +46,30 @@ class SolverCommon {
  */
 static bool copySparseToKINSOL(const SparseMatrix& smj, SUNMatrix JJ, const int& size, sunindextype * lastRowVals);
 
+/**
+ *
+ * @brief propagate the matrix structure change to KINSOL structure
+ *
+ * @param smj Sparse matrix to copy to the KINSOL structure
+ * @param JJ KINSOL structure where to copy the matrix
+ * @param size size of the square matrix (nb columns)
+ * @param lastRowVals pointer to the latest value of the previous matrix
+ * @param LS linear solver pointer
+ * @param linearSolverName name of the linear solver name (KLU / NICSLU)
+ * @param log @b true if a log should be added if a complete re-initialization is done
+ */
+static void propagateMatrixStructureChangeToKINSOL(const SparseMatrix& smj, SUNMatrix JJ, const int& size,
```

Comment:
Please pass the matrix and solver as a reference to avoid useless copy

Reply:
Yes, done